tests/ethpm/backends/test_ipfs_backends.py

```diff
@@ -27,6 +33,19 @@ def test_ipfs_and_infura_gateway_backends_fetch_uri_contents(
         assert contents_dict["package_name"] == "safe-math-lib"
 
 
+def test_local_ipfs_backend(monkeypatch):
+    uri = "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGV"
+    ipfs_client = ipfsapi.Client("localhost", "8080")
+
+    def mockreturn(path):
```

Comment:
Yeah, that's one way to do what I was thinking.  More specifically, I had imagine something like this which seems a bit more in line with what was discussed in the article:
```python
class FakeClient:
    def cat(self, ipfs_hash):
        return b"pragma"

backend = LocalIPFSBAckend(FakeClient())
...
```
What you did is effectively the same but just a bit more complicated.

You could also maybe do something like this which would also verify that the correct argument was given to `cat`:
```python
class FakeClient:
    def cat(self, ipfs_hash):
        return (ipfs_hash + "_contents").encode("utf-8")

backend = LocalIPFSBAckend(FakeClient())
...
assert contents == b"Qme4otp..._contents"
```